js/src/wrappers/ai-sdk/ai-sdk.ts

```diff
@@ -133,9 +167,18 @@ const makeGenerateTextWrapper = (
   generateText: any,
   aiSDK?: any,
 ) => {
-  const wrapper = async function (params: any) {
+  const wrapper = async function (allParams: any) {
+    // Extract span_info from params (used by Braintrust-managed prompts)
+    const { span_info, ...params } = allParams;
+    const {
+      metadata: spanInfoMetadata,
+      name: spanName,
+      spanAttributes: spanInfoAttrs,
+    } = span_info ?? {};
+
     const { model: initialModel, provider: initialProvider } =
       serializeModelWithProvider(params.model);
+    const effectiveProvider = params.model?.provider || initialProvider;
```

Comment:
maybe this means that serializeModelWithProvider should receive all of the params to keep the output consistent without the need to do the manual params.model?.provider check